File: CPP/26. Remove Duplicates from Sorted Array.cpp
#include<bits/stdc++.h>
using namespace std;

int removeDuplicates(vector<int>& nums)
{
    int i = 1;
    int prev = nums[0];
    for(int j = 1; j < nums.size() ; j++)
    {
        if(nums[j] != prev)
        {
            prev = nums[j];
            nums[i] = nums[j];
            i++;
        }
    }
    return i;
}

int main()
{
    int n;
    cin >> n;

    vector<int> arr(n);
    for(int i=0; i<n; i++) cin >> arr[i];

    int ans = removeDuplicates(arr);
    cout << ans << endl;


}

File: Python/26. Remove Duplicates from Sorted Array.py
def removeDuplicates(nums) -> int:
    i = 1
    prev = nums[0]
    n = len(nums)
    for j in range(1, n):
        if prev != nums[j]:
            prev = nums[j]
            nums[i] = nums[j]
            i += 1
    return i


def main():
    # li = list(map(int, input().split()))
    li = [int(x) for x in input().split()]
    ans = removeDuplicates(li)
    print(ans)


if __name__ == '__main__':
    main()
